File: leaf-system/src/main/java/com/leaf/domain/LeafUser.java
package com.leaf.domain;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;


/**
 * 用户登录信息表(LeafUser)实体类
 *
 * @author jhlz
 * @version 1.0.0
 */
@Entity
public interface LeafUser extends ID, BaseEntity {

    /**
     * 用户名
     */
    String username();

    /**
     * 密码
     */
    String password();

    /**
     * 昵称
     */
    String nickName();

    /**
     * 电话
     */
    @Nullable
    String phone();

    /**
     * 邮箱
     */
    @Nullable
    String email();

    /**
     * 头像地址
     */
    String avatar();

    /**
     * 最后一次登录时间
     */
    LocalDateTime lastLoginTime();

    /**
     * 状态：0 正常，1 禁用
     */
    int status();

    /**
     * 用户详情 ID
     */
    @Nullable
    @IdView
    Long userDetailId();

    /**
     * 部门 id
     */
    @IdView("dept")
    Long deptId();

    /**
     * 角色 ids
     */
    @IdView("roles")
    List<Long> roleIds();

    @Nullable
    @OneToOne
    // @JoinColumn(name = "user")
    LeafUserDetail userDetail();

    @Nullable
    @ManyToOne
    LeafDept dept();

    @ManyToMany
    @JoinTable(
            name = "leaf_user_role",
            joinColumnName = "id",
            inverseJoinColumnName = "id"
    )
    List<LeafRole> roles();

    // @ManyToMany
    // @JoinTable(
    //         name = "leaf_user_post",
    //         joinColumnName = "user_id",
    //         inverseJoinColumnName = "post_id"
    // )
    // List<LeafPost> postList();

}

File: src/main/kotlin/com/jhlz/domain/LeafUser.kt
package com.jhlz.domain

import org.babyfish.jimmer.sql.*

/**
 * 用户登录信息表(LeafUser)实体类
 *
 * @author jhlz
 * @version 1.0.0
 */
@Entity
interface LeafUser : BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    /**
     * 用户名
     */
    @Key
    val username: String

    /**
     * 密码
     */
    val password: String

    /**
     * 昵称
     */
    val nickname: String

    /**
     * 电话
     */
    val phone: String?

    /**
     * 邮箱
     */
    val email: String?

    /**
     * 头像地址
     */
    val avatar: String

    /**
     * 状态：0 正常，1 禁用
     */
    val status: Int

    /**
     * 部门 id
     */
    @IdView
    val deptId: Long

    /**
     * 用户详情 ID
     */
    val userDetailId: Long?

    @ManyToMany
    @JoinTable(
        name = "leaf_user_role",
        joinColumnName = "user_id",
        inverseJoinColumnName = "role_id",
    )
    val roles: List<LeafRole>

    @ManyToOne
    val dept: LeafDept
}

